examples/dedicated_vmseries_and_autoscale/variables.tf

```diff
@@ -0,0 +1,285 @@
+  - `ssl_profiles` : (optional) a map of SSL profiles that can be later on referenced in HTTPS listeners by providing a name of the profile in the `ssl_profile_name` property
+
+  EOF
+  default     = {}
+}
+
+
+variable "application_insights" {
+  description = <<-EOF
+  A map defining Azure Application Insights. There are three ways to use this variable:
+
+  * when the value is set to `null` (default) no AI is created
+  * when the value is a map containing `name` key (other keys are optional) a single AI instance will be created under the name that is the value of the `name` key
+  * when the value is an empty map or a map w/o the `name` key, an AI instance per each VMSeries VM will be created. All instances will share the same configuration. All instances will have names corresponding to their VM name.
+
+  Names for all AI instances are prefixed with `var.name_prefix`.
+
+  Properties supported (for details on each property see [modules documentation](../modules/application_insights/README.md)):
+
+  - `name` : (optional, string) a name of a single AI instance
+  - `workspace_mode` : (optional, bool) defaults to `true`, use AI Workspace mode instead of the Classical (deprecated)
+  - `workspace_name` : (optional, string) defaults to AI name suffixed with `-wrkspc`, name of the Log Analytics Workspace created when AI is deployed in Workspace mode
+  - `workspace_sku` : (optional, string) defaults to PerGB2018, SKU used by WAL, see module documentation for details
+  - `metrics_retention_in_days` : (optional, number) defaults to current Azure default value, see module documentation for details
+
+  Example of an AIs created per VM, in Workspace mode, with metrics retention set to 1 year:
+  ```
+  vmseries = {
+    'vm-1' = {
+      ....
+    }
+    'vm-2' = {
+      ....
+    }
+  }
+
+  application_insights = {
+    metrics_retention_in_days = 365
+  }
+  ```
+  EOF
+  default     = null
+  type        = map(string)
+}
+
+
+
+### GENERIC VMSERIES
+variable "vmseries_version" {
+  description = "VM-Series PAN-OS version - list available with `az vm image list -o table --all --publisher paloaltonetworks`"
+  type        = string
+}
+
+variable "vmseries_vm_size" {
+  description = "Azure VM size (type) to be created. Consult the *VM-Series Deployment Guide* as only a few selected sizes are supported."
+  type        = string
+}
+
+variable "vmseries_sku" {
+  description = "VM-Series SKU - list available with `az vm image list -o table --all --publisher paloaltonetworks`"
+  default     = "byol"
+  type        = string
+}
+
+variable "vmseries_username" {
+  description = "Initial administrative username to use for all systems."
+  default     = "panadmin"
+  type        = string
+}
+
+variable "vmseries_password" {
+  description = "Initial administrative password to use for all systems. Set to null for an auto-generated password."
+  default     = null
+  type        = string
+}
+
+variable "vmss" {
+  description = <<-EOF
```

Comment:
Could we please prepare description for `vmss` variable?

Reply:
good point :D

Reply:
done